### PR TITLE
Adding a sample on how to use PoP (Proof of Possession)

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose
Adds a folder showing how to use Proof of Possession

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code

```
Follow Readme.md
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 1.1 Desktop app calls Web API - PoP/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
+{
+  "Sample": {
+    "Title": "A WPF application that calls a Web API running on ASP.NET Core 2.0 protected by Azure AD using OAuth 2.0 access tokens. ",
```
Comment (bgavrilMS) [inline 1.1 Desktop app calls Web API - PoP/AppCreationScripts/sample.json]: It's a console app now no?

Comment (bgavrilMS) [reply 1.1 Desktop app calls Web API - PoP/AppCreationScripts/sample.json]: Also the WPF screenshots are no longer relevant.

File: 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Security.Cryptography;
+using Microsoft.Identity.Client;
+
+namespace TodoListClient
+{
+    static class TokenCacheHelper
```
Comment (bgavrilMS) [inline 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs]: You should add a comment that this helper is ok for Windows only and point to the extension for Linux and Mac.

File: 1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                .WithExperimentalFeatures() // Needed for PoP
+                .Build();
+            TokenCacheHelper.EnableSerialization(app.UserTokenCache);
+
+            AuthenticationResult result;
+
+            HttpClient httpClient = new HttpClient();
+            HttpResponseMessage response;
+
+            while (true)
```
Comment (bgavrilMS) [inline 1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs]: This method is too long - can you split it into 2 private methods, maybe "WriteItem" and "ReadItem" ?

File: 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs (hunk 2 above)
Comment (bgavrilMS) [reply 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs]: (I know POP is only available on .net classic for now..)

File: 1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs (hunk 3 above)
Comment (bgavrilMS) [reply 1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs]: Or better yet - create a little menu like here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/a9f1175dd747777bcacea5063c8211f94f1f0ec0/tests/devapps/NetFxConsoleTestApp/Program.cs#L159

Comment (bgavrilMS) [reply 1.1 Desktop app calls Web API - PoP/TodoListClient/Program.cs]: Let me know if you want me to do this for you.

File: 1.1 Desktop app calls Web API - PoP/AppCreationScripts/sample.json (hunk 1 above)
Comment (jmprieur) [reply 1.1 Desktop app calls Web API - PoP/AppCreationScripts/sample.json]: I could re-introduce the WPF application now. Wondering about that.
Right the screenshots need to be updated

File: 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs (hunk 2 above)
Comment (jmprieur) [reply 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs]: Yes. BTW @TiagoBrenck has created a cross platform sample for desktop; https://github.com/Azure-Samples/ms-identity-dotnet-desktop-tutorial/pull/4

Comment (jmprieur) [reply 1.1 Desktop app calls Web API - PoP/TodoListClient/TokenCacheHelper.cs]: Thanks @bgavrilMS 
Done

**Pull request merged**